lapce-data/src/document.rs

```diff
@@ -1319,7 +1319,11 @@ impl Document {
             Movement::FirstNonBlank => {
                 let line = self.buffer.line_of_offset(offset);
                 let new_offset = self.buffer.first_non_blank_character_on_line(line);
-                (new_offset, Some(ColPosition::FirstNonBlank))
+                if offset == new_offset {
```

Comment:
Maybe it should be ```if offset <= new_offset``` ?

Reply:
So, the reason for this version is that if your line is
```
<|>    blah
```
and you press home, the correct behaviour is to make your cursor move to:
```
    <|>blah
```
At least, that's what vscode does.
Basically, pressing home should cycle between the two states, defaulting to the first non-whitespace character.

Reply:
I just tested VSCode,

If the cursor is at the left of the first non-whitespace character, Home gets to the line beginning?

Reply:
Ah, great. I'll implement that version then